packages/web3-providers/src/RedPacket/index.ts

```diff
@@ -121,13 +121,14 @@ class RedPacketAPI implements RedPacketBaseAPI.Provider<ChainId, SchemaType> {
         transactions: Array<Transaction<ChainId, SchemaType>> | undefined,
         senderAddress: string,
     ): NftRedPacketJSONPayload[] {
-        if (!transactions) return EMPTY_LIST
+        if (!transactions) return []
```

Comment:
If the outcome needs to be mutable at any point, it's advisable to refactor the code.

Reply:
Agree.

But it's not explicitly immutable here, besides, `return transactions.flatMap()` is mutable, keeping all returns consistent. We cannot restrict how others use the result.

And I think `EMPTY_LIST` should be preferred to use in hooks, props and some other places need references comparison.